core/arch/x86/x86.asm

```diff
@@ -2284,10 +2284,96 @@ GLOBAL_LABEL(get_ymm_caller_saved:)
         ret
         END_FUNC(get_ymm_caller_saved)
 
-/* void hashlookup_null_handler(void)
- * PR 305731: if the app targets NULL, it ends up here, which indirects
- * through hashlookup_null_target to end up in an ibl miss routine.
+/* void get_zmm_caller_saved(byte *zmm_caller_saved_buf)
+ *   stores the values of zmm0 through zmm31 consecutively into zmm_caller_saved_buf.
+ *   zmm_caller_saved_buf need not be 64-byte aligned.
+ *   caller must ensure that the underlying processor supports AVX-512!
  */
+        DECLARE_FUNC(get_zmm_caller_saved)
+GLOBAL_LABEL(get_zmm_caller_saved:)
+        mov      REG_XAX, ARG1
+       /* i#441: some compilers like gcc 4.3 and VS2005 do not know "vmovdqu"
```

Comment:
Comment seems really stale: we don't support VS2005, or 2008, or 2010.  Not sure anybody has built with gcc 4.3 in years.

Reply:
Moreover, I think it is misleading. Because AVX/AVX2/AVX-512 is not default, it always needs the architectural switch. I have modified the entire comment both for the ymm and zmm version.